driver/lib/sessions/ios.ts

```diff
@@ -22,11 +25,22 @@ export const startIOSSession = async (caps) => {
   const observatoryWsUri = getObservatoryWsUri(iosdriver);
   return Promise.all([
     iosdriver,
-    connectSocket(observatoryWsUri),
+    connectSocket(await observatoryWsUri),
   ]);
 };
 
-export const getObservatoryWsUri = (proxydriver) => {
+export const getObservatoryWsUri = async (proxydriver) => {
   const urlObject = processLogToGetobservatory(proxydriver.logs.syslog.logs);
+  const { udid, realDevice } = proxydriver.opts;
+  if (realDevice) {
+    // @todo check if `brew install usbmuxd` is needed
+    log.info(`Running on iOS real device, doing "iproxy" now`);
+    const cmd = `iproxy ${urlObject.port} ${urlObject.port} ${udid}`;
+    log.debug(cmd);
+    await execPromise(cmd);
```

Comment:
the permanent blocking of processes

Reply:
@YueChen-C Do you mean that await and promise needs to be remove. And use below.
child_process.exec(command)

Reply:
I may use spawn here without await and wait for a String "waiting for connection" if it resolve I can send the URL.

Reply:
good solution

Reply:
Let me apply those and release a bump version